src/domain/email/email.repository.ts

```diff
@@ -38,10 +38,15 @@ export class EmailRepository implements IEmailRepository {
 
     const verificationCode = codeGenerator();
 
+    // Pads the final verification code to 6 characters
+    // The generated code might have less than 6 digits so the version to be
+    // validated against should account with the leading zeroes
+    const paddedVerificationCode = verificationCode.toString().padStart(6, '0');
```

Comment:
Nitpick: should we move this logic and make `code-generator.ts` to return a padded string? I guess the response depends on the foreseen use cases for the code generator. So I left this up to you, just commenting for your consideration.

Reply:
The main reason why I didn't add this logic there was mostly to leave `code-generator.ts` to deal with numbers only.

So the padding part could be seen as an extra step if we want to store it in the DB as such thus having this logic in the repository.

But no other reasons behind having this logic here. We can also see how it evolves. One option is also to merge the `code-generator` with the repository.